src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1220,49 +1223,275 @@ static bool addTriggeredSubscriptions
+                  getSubscribeContextCollectionName(tenant).c_str(),
+                  query.toString().c_str()));
+
+  /* Do the query */
+  auto_ptr<DBClientCursor> cursor;
+  try
+  {
+    connection      = getMongoConnection();
+    cursor = connection->query(getSubscribeContextCollectionName(tenant).c_str(), query);
+
+    /*
+     * We have observed that in some cases of DB errors (e.g. the database daemon is down) instead of
+     * raising an exception, the query() method sets the cursor to NULL. In this case, we raise the
+     * exception ourselves
+     */
+    if (cursor.get() == NULL)
+    {
+      throw DBException("Null cursor from mongo (details on this is found in the source code)", 0);
+    }
+    releaseMongoConnection(connection);
+
+    LM_I(("Database Operation Successful (%s)", query.toString().c_str()));
+  }
+  catch (const DBException &e)
+  {
+    releaseMongoConnection(connection);
+    err = std::string("collection: ") + getSubscribeContextCollectionName(tenant).c_str() +
+               " - query(): " + query.toString() +
+               " - exception: " + e.what();
+    LM_E(("Database Error (%s)", err.c_str()));
+    return false;
+  }
+  catch (...)
+  {
+    releaseMongoConnection(connection);
+    err = std::string("collection: ") + getSubscribeContextCollectionName(tenant).c_str() +
+               " - query(): " + query.toString() +
+               " - exception: " + "generic";
+    LM_E(("Database Error (%s)", err.c_str()));
+    return false;
+  }
+
+  /* For each one of the subscriptions found, add it to the map (if not already there) */
+  while (cursor->more())
+  {
+    BSONObj      sub      = cursor->next();
+    BSONElement  idField  = sub.getField("_id");
+
+    //
+    // BSONElement::eoo returns true if 'not found', i.e. the field "_id" doesn't exist in 'sub'
+    //
+    // Now, if 'sub.getField("_id")' is not found, if we continue, calling OID() on it, then we get
+    // an exception and the broker crashes.
+    //
+    if (idField.eoo() == true)
+    {
+      LM_E(("Database Error (error retrieving _id field in doc: %s)", sub.toString().c_str()));
+      continue;
+    }
+
+    std::string subIdStr = idField.OID().toString();
+
+    if (subs.count(subIdStr) == 0)
+    {
+      LM_T(LmtMongo, ("adding subscription: '%s'", sub.toString().c_str()));
+
+      long long throttling       = sub.hasField(CSUB_THROTTLING) ? sub.getField(CSUB_THROTTLING).numberLong() : -1;
+      long long lastNotification = sub.hasField(CSUB_LASTNOTIFICATION) ? sub.getIntField(CSUB_LASTNOTIFICATION) : -1;
+
+      TriggeredSubscription* trigs = new TriggeredSubscription
+        (
+          throttling,
+          lastNotification,
+          sub.hasField(CSUB_FORMAT) ? stringToFormat(STR_FIELD(sub, CSUB_FORMAT)) : XML,
+          STR_FIELD(sub, CSUB_REFERENCE),
+          //subToAttributeList(sub)); signature changed in TriggeredSubscription() constructor in 0.24.0
```

Comment:
Yes, but not mine.
I saw it and I thought it was yours, so I left it :-)
Removing it.